Riot/Modules/Spaces/SpaceMenu/SpaceMenuViewModel.swift

```diff
@@ -75,29 +75,52 @@ class SpaceMenuViewModel: SpaceMenuViewModelType {
         guard let space = self.session.spaceService.getSpace(withId: self.spaceId), let displayName = space.summary?.displayname else {
             return
         }
+
+        var message = VectorL10n.leaveSpaceMessage(displayName)
+
+        if let roomState = space.room.dangerousSyncState, let powerLevels = roomState.powerLevels {
+            let powerLevel = powerLevels.powerLevelOfUser(withUserID: self.session.myUserId)
+            let roomPowerLevel = RoomPowerLevelHelper.roomPowerLevel(from: powerLevel)
+            if roomPowerLevel == .admin {
+                message += "\n\n" + VectorL10n.leaveSpaceMessageAdminWarning
+            }
+        }
         
-        let alert = UIAlertController(title: nil, message: VectorL10n.leaveSpaceMessage(displayName), preferredStyle: .alert)
+        let alert = UIAlertController(title: VectorL10n.leaveSpaceTitle(displayName), message: message, preferredStyle: .alert)
 
-        alert.addAction(UIAlertAction(title: VectorL10n.leave, style: .destructive, handler: { [weak self] action in
+        alert.addAction(UIAlertAction(title: VectorL10n.leaveSpaceOnlyAction, style: .default, handler: { [weak self] action in
```

Comment:
We should avoid having UI in view models. They should be independent from screen representation. In this case it's better to had new ViewState(s) and ViewAction(s). We can iterate later on that.